rust/operator-binary/src/config/jvm.rs

```diff
@@ -60,69 +63,232 @@ pub fn jvm_config(
+            "-XX:ReservedCodeCacheSize=512M".to_owned(),
+            "-XX:PerMethodRecompilationCutoff=10000".to_owned(),
+            "-XX:PerBytecodeRecompilationCutoff=10000".to_owned(),
+            "-Djdk.attach.allowAttachSelf=true".to_owned(),
+            "-Djdk.nio.maxCachedBufferSize=2000000".to_owned(),
+            "-Dfile.encoding=UTF-8".to_owned(),
+            "-XX:+EnableDynamicAgentLoading".to_owned(),
+            "-XX:+UnlockDiagnosticVMOptions".to_owned(),
+            "-XX:G1NumCollectionsKeepPinned=10000000".to_owned(),
+        ]),
         // Copied from https://trino.io/docs/455/installation/deployment.html#jvm-config
-        "455" => Ok(formatdoc!(
-            "-server
-            -Xms{heap}
-            -Xmx{heap}
-
-            -XX:InitialRAMPercentage=80
-            -XX:MaxRAMPercentage=80
-            -XX:G1HeapRegionSize=32M
-            -XX:+ExplicitGCInvokesConcurrent
-            -XX:+ExitOnOutOfMemoryError
-            -XX:+HeapDumpOnOutOfMemoryError
-            -XX:-OmitStackTraceInFastThrow
-            -XX:ReservedCodeCacheSize=512M
-            -XX:PerMethodRecompilationCutoff=10000
-            -XX:PerBytecodeRecompilationCutoff=10000
-            -Djdk.attach.allowAttachSelf=true
-            -Djdk.nio.maxCachedBufferSize=2000000
-            -Dfile.encoding=UTF-8
-            # Allow loading dynamic agent used by JOL
-            -XX:+EnableDynamicAgentLoading
-
-            -Djavax.net.ssl.trustStore={STACKABLE_CLIENT_TLS_DIR}/truststore.p12
-            -Djavax.net.ssl.trustStorePassword={STACKABLE_TLS_STORE_PASSWORD}
-            -Djavax.net.ssl.trustStoreType=pkcs12
-            -Djava.security.properties={RW_CONFIG_DIR_NAME}/{JVM_SECURITY_PROPERTIES}
-            ",
-        )),
+        "455" => Ok(vec![
+            "-XX:InitialRAMPercentage=80".to_owned(),
+            "-XX:MaxRAMPercentage=80".to_owned(),
+            "-XX:G1HeapRegionSize=32M".to_owned(),
+            "-XX:+ExplicitGCInvokesConcurrent".to_owned(),
+            "-XX:+ExitOnOutOfMemoryError".to_owned(),
+            "-XX:+HeapDumpOnOutOfMemoryError".to_owned(),
+            "-XX:-OmitStackTraceInFastThrow".to_owned(),
+            "-XX:ReservedCodeCacheSize=512M".to_owned(),
+            "-XX:PerMethodRecompilationCutoff=10000".to_owned(),
+            "-XX:PerBytecodeRecompilationCutoff=10000".to_owned(),
+            "-Djdk.attach.allowAttachSelf=true".to_owned(),
+            "-Djdk.nio.maxCachedBufferSize=2000000".to_owned(),
+            "-Dfile.encoding=UTF-8".to_owned(),
+            "-XX:+EnableDynamicAgentLoading".to_owned(),
+        ]),
         _ => TrinoVersionNotSupportedSnafu {
-            version: resolved_product_image.product_version.clone(),
+            version: product_version,
         }
         .fail(),
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use indoc::indoc;
+    use stackable_trino_crd::{TrinoCluster, TrinoRole};
+
+    use super::*;
+
+    #[test]
+    fn test_jvm_config_defaults() {
```

Comment:
I feel this is a lot of code/yamls (that may change every other version bump) for not much testing.

Not sure we need the tests at all. I think the overrides are tested properly in op-rs. If you want like a sanity/safety check i would skim it down a little (e.g. work on role and smaller defaults) and not work on the (changing) trino configs?

Happy to discuss if i miss something :)

Reply:
I personally was very happy about the tests (in op-rs and here) during development.
I think the maintenance effort is pretty low, just let the test run and copy/paste the actual output into the expected one. This applies e.g. for a new Trino version.
Also this tests comments and switching between different Trino versions, which most operators don't do.

I'm totally happy to have much easier or no tests in the other operators, but would prefer to not throw away the effort I already did and helped me developing and gaining confidence.